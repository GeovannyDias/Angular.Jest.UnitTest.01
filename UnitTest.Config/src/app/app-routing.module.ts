import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { CharizardComponent } from './basic/charizard/charizard.component';
import { CounterComponent } from './basic/counter/counter.component';
import { CouterRouteComponent } from './basic/couter-route/couter-route.component';
import { FatherComponent } from './basic/father/father.component';

const routes: Routes = [
  {
    path: 'basic/couter',
    component: CounterComponent
  },
  {
    path: 'basic/couter/:initial',
    component: CouterRouteComponent
  },
  {
    path: 'basic/charizard',
    component: CharizardComponent
  },
  {
    path: 'basic/father',
    component: FatherComponent
  },
  {
    path: '**',
    redirectTo: 'basic/couter'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
