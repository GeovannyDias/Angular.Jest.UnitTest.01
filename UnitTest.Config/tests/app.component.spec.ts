import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../src/app/app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  // it() es el alias de test(), misma funcionalidad (legibilidad)

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    // const app = fixture.componentInstance;
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  it(`should have as title 'UnitTest.Config'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    // const app = fixture.componentInstance;
    const component = fixture.componentInstance;
    // const method = component.increase();
    expect(component.title).toEqual('UnitTest.Config');
  });

  // it('should render title', () => {
  //   const fixture = TestBed.createComponent(AppComponent);
  //   fixture.detectChanges();
  //   const compiled = fixture.nativeElement as HTMLElement;
  //   expect(compiled.querySelector('.content span')?.textContent).toContain('UnitTest.Config app is running!');
  // });

  it('should render title', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance; // Hacer global

    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    const h1 = compiled.querySelector('h1');

    expect(h1?.textContent).toContain(component.title);
  });


  // SNAPSHOT (Copy render) - (Update press `u` or delete folder __snapshots__)

  test('Should do match with Snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges(); // Refresca los cambios
    const compiled = fixture.nativeElement as HTMLElement; // Componente compilado

    // expect(compiled).toMatchInlineSnapshot(); // Crea el componente en este fichero
    expect(compiled).toMatchSnapshot(); // Crea una replica del componente el la carpeta __snapshots__
  });


});
