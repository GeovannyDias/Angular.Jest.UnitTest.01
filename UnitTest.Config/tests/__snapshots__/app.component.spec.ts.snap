// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Should do match with Snapshot 1`] = `
<div
  id="root3"
>
  <h1>
    UnitTest.Config
  </h1>
  <hr />
  <router-outlet />
  
</div>
`;
